sdk/schemaregistry/Azure.Data.SchemaRegistry/src/Serialization/SchemaRegistrySerializer.cs

```diff
@@ -0,0 +1,550 @@
+        ///   will contain more information.
+        /// </exception>
+        public async ValueTask<object> DeserializeAsync(
+            MessageContent content,
+            Type dataType,
+            CancellationToken cancellationToken = default)
+            => await DeserializeMessageDataInternalAsync(content.Data, dataType, content.ContentType, true, cancellationToken).ConfigureAwait(false);
+
+        private async ValueTask<object> DeserializeMessageDataInternalAsync(
+            BinaryData data,
+            Type dataType,
+            Azure.Core.ContentType? contentType,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(data, nameof(data));
+            Argument.AssertNotNull(contentType, nameof(contentType));
+
+            string[] contentTypeArray = contentType.ToString().Split('+');
+            if (contentTypeArray.Length != 2 || contentTypeArray[0] != _mimeType)
+            {
+                throw new FormatException($"Content type was not in the expected format of '{_mimeType}+schema-id', where 'schema-id' " +
+                                          "is the Schema Registry schema ID.");
+            }
+
+            string schemaId = contentTypeArray[1];
+
+            if (async)
+            {
+                return await DeserializeInternalAsync(data, dataType, schemaId, true, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                return DeserializeInternalAsync(data, dataType, schemaId, false, cancellationToken).EnsureCompleted();
+            }
+        }
+
+        private async ValueTask<object> DeserializeInternalAsync(
+            BinaryData data,
+            Type dataType,
+            string schemaId,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            Argument.AssertNotNull(schemaId, nameof(schemaId));
+
+            string schemaDefinition;
+            try
+            {
+                // Attempt to get the schema definition using the schema Id from the Content Type
+                if (async)
+                {
+                    schemaDefinition = await GetSchemaByIdAsync(schemaId, true, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    schemaDefinition = GetSchemaByIdAsync(schemaId, false, cancellationToken).EnsureCompleted();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"An error occurred while attempting to retrieve the schema with id {schemaId}.", ex);
+            }
+
+            var deserializer = _serializerOptions.Serializer;
+            object objectToReturn;
+            try
+            {
+                // Attempt to deserialize
+                var dataStream = data.ToStream();
```

Comment:
Does the stream need to be wrapped in an using? Haven't look at the implementation details of `BinaryData.ToStream`